### PR TITLE
WIP: Add Registry Metadata Utility

### PR DESCRIPTION
**Pull request by Zombiefruit**

### CODE REVIEW

**Comment by shlomitc** [comment]
@Zombiefruit - can you please add some explanation what exactly it does?
If it's not yet ready to merge, please add a WIP to the title.

**Comment by nadavwix** [comment]
this looks fine.

@shlomitc  this is a meta-data repo for components.

it allows running automated tools (such as an auto SSRTool on all components)


@Zombiefruit maybe add the auto SSR tool to the same PR

File: packages/wix-ui-test-utils/src/meta-data-tools/meta-data-tools.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    return this.metaData.get(comp)!;
+  },
+  clean () {
+    this.metaData = new Map();
+  }
+};
+
+export default MetaDataTools;
```
Comment (shlomitc) [inline packages/wix-ui-test-utils/src/meta-data-tools/meta-data-tools.ts]: avoid using `default` export

File: packages/wix-ui-test-utils/src/meta-data-tools/meta-data-tools.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/* tslint:disable:no-invalid-this */
+import {ComponentClass} from 'react';
+import {MetaDataToolsDefinition, MetaDataDefinition, Simulation} from './types';
+
+export class MetaData implements MetaDataDefinition {
```
Comment (shlomitc) [inline packages/wix-ui-test-utils/src/meta-data-tools/meta-data-tools.ts]: Why do you need it? Where is it in use?

File: packages/wix-ui-test-utils/package.json [hunk 3]
```diff
@@ -56,7 +57,7 @@
     "react-test-renderer": "^16.3.1",
     "ts-jest": "^22.0.3",
     "typescript": "^2.6.2",
-    "yoshi": "^1.2.0"
+    "yoshi": "^2.6.0"
```
Comment (shlomitc) [inline packages/wix-ui-test-utils/package.json]: can you make it as a separate PR? it's a package used in many places so I prefer it to be updated without the connection to your features

File: packages/wix-ui-test-utils/src/auto-ssr/index.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+import MetaDataTools from './auto-ssr';
+export default MetaDataTools;
```
Comment (shlomitc) [inline packages/wix-ui-test-utils/src/auto-ssr/index.ts]: don't use default imports please

File: packages/wix-ui-test-utils/src/meta-data-tools/index.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import MetaDataTools, {MetaData} from './meta-data-tools';
+export {MetaData};
+export default MetaDataTools;
```
Comment (shlomitc) [inline packages/wix-ui-test-utils/src/meta-data-tools/index.ts]: don't use default imports please

File: packages/wix-ui-test-utils/src/auto-ssr/auto-ssr.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
```
Comment (shlomitc) [inline packages/wix-ui-test-utils/src/auto-ssr/auto-ssr.tsx]: regarding this tool. what is the plan for it? because if we do also hidration tests (opening a browser), I think it's redundant to run it as a separate step in the process. the server should not serve the file if some errors occur
